### PR TITLE
refactor: deprecate old s3 and use use s3_v2 as default

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

as title

resolve https://github.com/risingwavelabs/risingwave/issues/16645

replace `s3` with `s3_v2` in the frontend so we can migrate the default to `s3`

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

please remove the prev s3 source part and rename s3_v2 into s3

</details>

### CODE REVIEW

**Review by xxchan — COMMENTED** [review]
LGTM. What about deprecating `s3_v2` at the same time?

i.e.,
- Backend: `s3` and `s3_v2` both kept, and nothing changed
- Frontend: `s3` will be rewritten to `s3_v2`, `s3_v2` will be rejected.

**Comment by tabVersion** [comment]
> LGTM. What about deprecating `s3_v2` at the same time?
> 
> i.e.,
> 
> * Backend: `s3` and `s3_v2` both kept, and nothing changed
> * Frontend: `s3` will be rewritten to `s3_v2`, `s3_v2` will be rejected.

That's what I am thinking about. Some old user may follow their experience to proceed with `s3_v2`, but we tell them `s3_v2` is deprecated and migrating to `s3` seems strange. I prefer keeping both `s3` and `s3_v2`.

File: src/frontend/src/handler/create_source.rs [hunk 1]
```diff
@@ -1117,10 +1116,18 @@ pub fn validate_compatibility(
...
         .get_connector()
         .ok_or_else(|| RwError::from(ProtocolError("missing field 'connector'".to_string())))?;
 
+    if connector == S3_CONNECTOR {
+        // S3 connector is deprecated, use OPENDAL_S3_CONNECTOR instead
+        // do s3 -> s3_v2 migration
+        let entry = props.get_mut(UPSTREAM_SOURCE_KEY).unwrap();
+        *entry = OPENDAL_S3_CONNECTOR.to_string();
+        connector = OPENDAL_S3_CONNECTOR.to_string();
+    }
```
Comment (fuyufjh) [inline src/frontend/src/handler/create_source.rs:1129]: Will these lines work for an existing job?

Comment (tabVersion) [reply src/frontend/src/handler/create_source.rs:1129]: Only for new jobs. The existing jobs do not go through frontend again.

Comment (fuyufjh) [reply src/frontend/src/handler/create_source.rs:1129]: xxchan's proposal https://github.com/risingwavelabs/risingwave/pull/17963#pullrequestreview-2225725091 sounds the most reasonable to me, that is, rejecting the `s3_v2` in frontend, but allow the existing ones. Can we do that?

Comment (xxchan) [reply src/frontend/src/handler/create_source.rs:1129]: BTW, I think the changes here are rewriting `s3` -> `s3_v2`, which is necessary.

Comment (tabVersion) [reply src/frontend/src/handler/create_source.rs:1129]: > rejecting the s3_v2 in frontend, but allow the existing ones. Can we do that?

Technically, yes.
My concern is from the product side. Rejecting a higher version but accepting a lower one seems strange. And what if we evolve a new s3 source, we'd call it s3_v2 or s3_v3?

**Comment by tabVersion** [comment]
adjust the behavior to ban s3_v2

File: src/frontend/src/handler/create_source.rs (hunk 1 above)
Comment (fuyufjh) [reply src/frontend/src/handler/create_source.rs:1138]: Leaving `s3_v2` as an alias sounds acceptable to me, but please make sure all the docs & test cases are updated to `s3`.

**Review by fuyufjh — APPROVED** [review]

**Pull request merged**